posthog/api/action.py

```diff
@@ -355,22 +361,33 @@ def calculate_trends(filter: Filter, team_id: int, actions: QuerySet) -> List[Di
     return entities_list
 
 
-def calculate_retention(filter: Filter, team: Team, start_entity: Optional[Entity] = None, total_days=11):
+def calculate_retention(filter: Filter, team: Team, period: str, start_entity: Optional[Entity] = None, total_time=11):
+    def _determineTimedelta(total_time: int, period: str) -> Union[timedelta, relativedelta]:
+        if period == "Hour":
+            return timedelta(hours=total_time)
+        elif period == "Week":
+            return timedelta(weeks=total_time)
+        else:
```

Comment:
This is a dangerous pattern, if someone enters `period="Month", total_time=1` for 1 month, they're actually getting 1 day.

Consider:
```
elif period == "Day":
    return timedelta(days=total_time)
else:
    raise ValueError(f"Period {period} is unsupported.")
```